internal/providers/manager/manager.go

```diff
@@ -101,12 +109,61 @@ func NewProviderManager(
 		}
 	}
 
-	return &providerManager{
+	return &classTracker{
 		classManagers: classes,
-		store:         store,
 	}, nil
 }
 
+type providerManager struct {
+	classTracker
+	store providers.ProviderStore
+}
+
+// NewProviderManager creates a new instance of ProviderManager
+func NewProviderManager(
+	store providers.ProviderStore,
+	classManagers ...ProviderClassManager,
+) (ProviderManager, error) {
+	classes, err := newClassTracker(classManagers...)
+	if err != nil {
+		return nil, fmt.Errorf("error creating class tracker: %w", err)
+	}
+
+	return &providerManager{
+		classTracker: *classes,
+		store:        store,
+	}, nil
+}
+
+func (p *providerManager) CreateFromConfig(ctx context.Context, providerClass db.ProviderClass, projectID uuid.UUID, name string, config json.RawMessage) (*db.Provider, error) {
+	manager, err := p.getClassManager(providerClass)
+	if err != nil {
+		return nil, fmt.Errorf("error getting class manager: %w", err)
+	}
+
+	provConfig, err := manager.GetConfig(ctx, config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting provider config: %w", err)
+	}
+
+	// This method feels like it should be part of the class manager?
+	providerDef, err := providers.GetProviderClassDefinition(string(providerClass))
+	if err != nil {
+		return nil, fmt.Errorf("error getting provider definition: %w", err)
+	}
+
+	providerParams := db.CreateProviderParams{
```

Comment:
The fact that we have to create a database-specific type here and pass it to the store is ugly, but I'm not sure if I have a good alternative here.

Reply:
I actually fixed this locally. I will push a new PR shortly.